File: Reverse the string/sol.py
// Method 1
class Solution:
    def reverseWord(self, s: str) -> str:
        e = ""
        for i in range(len(s) - 1, -1, -1):
            e += s[i]
        return e


// Method 2
class Solution:
    def reverseWord(self, s: str) -> str:
        str_list = list(s)
        i, j = 0, len(str_list) - 1
        while i < j:
            temp = str_list[i]
            str_list[i] = str_list[j]
            str_list[j] = temp
            i += 1
            j -= 1
        return ''.join(str_list)

File: Reverse the string/sol.cpp
// Method 1
class Solution
{
    public:
    string reverseWord(string str)
    {
    string reversed;
    for (int i = str.length() - 1; i >= 0; i--) {
        reversed += str[i];
    }
    return reversed;
    }
};

// Method 2
class Solution
{
    public:
    string reverseWord(string str)
    {
        int i=0;
        int j = str.length()-1;
        while(i < j){
            char temp = str[i];
            str[i] = str[j];
            str[j] = temp;
            i++;
            j--;
        }
        return str;
    }
};
